composer/trainer/trainer.py

```diff
@@ -2156,15 +2156,16 @@ def _iter_dataloader(self):
             dataloader_iter = itertools.islice(self.state.dataloader, int(self.state.dataloader_len))
 
         while True:
-            if marker is not None:
-                marker.start()
             try:
+                # [BEFORE/AFTER]_DATALOADER only runs while training
+                if trainer_mode == TrainerMode.TRAIN:
+                    self.engine.run_event(Event.BEFORE_DATALOADER)
                 batch = next(dataloader_iter)
             except StopIteration:
+                # [BEFORE/AFTER]_DATALOADER only runs while training
+                if trainer_mode == TrainerMode.TRAIN:
+                    self.engine.run_marker_only_event(Event.AFTER_DATALOADER)
                 break
```

Comment:
A little confused by this.. what happened to: 
https://github.com/mosaicml/composer/blob/dev/composer/trainer/trainer.py#L1513

(the run_event(AFTER_DATALOADER)), and why is the marker here? is it to more precisely measure the yield time?

Reply:
Added a comment -- sorry this was unclear. Basically, in the yield where we dont actually get anything (end of loop), we need to have a corresponding marker.finish() call